### DISCUSSION TITLE
How to assign a network interface to docker container in docker compose yaml

### DISCUSSION THREAD

**Discussion by NikuSD** [moby/moby#46111] [General]
I need to add a network interface to a docker container. Currently I am doing it as per below steps
1. Start the docker container.
2. create a softlink to the docker container namespace using `ln -sf /proc/<pid of container>/ns/net /var/run/netns/my_ns`
3. link the interface to container namespace using `ip link set eth2 netns my_ns`
4. set the interface up using `ip netns exec my_ns ip link set eth2 up`

This is working fine. But if docker container is restarted then I am not able to assign that interface again. That interface i.e. eth2 is not accessable for 100 seconds. After 100 seconds it becomes available and then only, I am able to follow the above steps.

Is there a way to assign the interface to the docker container using docker compose so that when docker container is restarted then it will get assign again.

**Comment by corhere** [comment]
No, there is no supported way to move a network link into a container like what you are doing manually. [What do you _really_ want to do?](https://xyproblem.info/)
